### DISCUSSION TITLE
How to change ui icons?

### DISCUSSION THREAD

**Discussion by iguii** [nvimdev/lspsaga.nvim#1217] [Q&A] [answered]
![image](https://github.com/nvimdev/lspsaga.nvim/assets/22847626/b9d6d746-0d69-42a2-bcd3-ec859f1045ab)
How can i change the icons that lspsaga uses for diagnosis?
I've tried this with my configuration but it doesn't seem to work:
```lua
require("lspsaga").setup({
  init_lsp_saga = {
      error_sign = "",
      warn_sign = "",
      hint_sign = "",
      infor_sign = "",
     border_style = "round",
  }
})
```

is there anyway I can make the icons change?

**Comment by glepnir** [comment]
you mean virtual text ? it not set by lspsaga `:help vim.diagnostic.config`

**Reply by iguii** [reply]
sorry, I meant this part. The icons that are next to line numbers.
<img width="55" alt="ScreenShot 2023-07-30 at 03 50 39@2x" src="https://github.com/nvimdev/lspsaga.nvim/assets/22847626/6fdb8a7d-1b01-4f38-b453-d6e5dc7a63b8">

**Reply by glepnir** [reply, marked as answer]
it also set by neovim not lspsaga you can define them like 

```lua
  local signs = {
    Error = ' ',
    Warn = ' ',
    Info = ' ',
    Hint = ' ',
  }
  for type, icon in pairs(signs) do
    local hl = 'DiagnosticSign' .. type
    vim.fn.sign_define(hl, { text = icon, texthl = hl, numhl = hl })
  end
```
